clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1011,7 +1056,11 @@ private boolean maybeDrainPendingCall(Call call, long now) {
             } catch (Throwable t) {
                 // Handle authentication errors while choosing nodes.
                 log.debug("Unable to choose node for {}", call, t);
-                call.fail(now, t);
+                if (t instanceof RetriableException) {
```

Comment:
Since we have this logic of checking what the type of `t` is in multiple places, we could consolidate this logic within `call.retry()`, since we are already passing the throwable instance. We could then also rename the method to be a bit more encompassing, like `attemptToRetry` or something.

Reply:
Added a new method acting as this switch. If the throwable is retriable, go retry(). Else, go fail().